devicedefender/source/DeviceDefender.cpp

```diff
@@ -100,37 +103,46 @@ namespace Aws
 
         int ReportTask::StartTask() noexcept
         {
-            if (this->GetStatus() == ReportTaskStatus::Ready || this->GetStatus() == ReportTaskStatus::Stopped)
+            int return_code = AWS_OP_ERR;
+            if (m_taskConfig != nullptr && !m_lastError &&
+                (this->GetStatus() == ReportTaskStatus::Ready || this->GetStatus() == ReportTaskStatus::Stopped))
             {
-
-                this->m_owningTask = aws_iotdevice_defender_v1_report_task(this->m_allocator, &this->m_taskConfig);
-
-                if (this->m_owningTask == nullptr)
+                if (AWS_OP_SUCCESS != aws_iotdevice_defender_task_create(
+                                          &m_owningTask,
+                                          this->m_taskConfig,
+                                          m_mqttConnection->GetUnderlyingConnection(),
+                                          aws_event_loop_group_get_next_loop(m_eventLoopGroup.GetUnderlyingHandle())))
                 {
                     this->m_lastError = aws_last_error();
                     aws_raise_error(this->m_lastError);
```

Comment:
trivial: there's no point in re-raising the error

Reply:
Removed